### ISSUE TITLE
uninlined-format-args suggests broken code for assert! macros in 2018 edition

### ISSUE THREAD

**Issue by inikulin** [rust-lang/rust-clippy#10234]
### Summary

`uninlined-format-args`, which is now enforced by default, complains about format literals passed to the `assert!` macros, but inlined format args don't work with them in 2018 edition.

### Reproducer

I tried this code:

```rust
fn main() {
    let hey = 42;
    
    assert!(false, "foo is {hey}");
}
```

I expected to see this happen: this to work as suggested by clippy

Instead, this happened: compilation warning + `foo is {hey}` being a panic message instead of `foo is 42`

### Version

```text
rustc 1.67.0 (fc594f156 2023-01-24)
binary: rustc
commit-hash: fc594f15669680fa70d255faec3ca3fb507c3405
commit-date: 2023-01-24
host: x86_64-apple-darwin
release: 1.67.0
LLVM version: 15.0.6
```

**Comment by taiki-e** [comment]
This was fixed in https://github.com/rust-lang/rust/commit/e5010c996e9cb9fa09fefb4e3d31608915869c78 (https://github.com/rust-lang/rust-clippy/pull/10055), but that patch is only included in 1.68 and later, not 1.67.

**Comment by djc** [comment]
In https://github.com/tokio-rs/tls/actions/runs/4021612870/jobs/6910637824 I noticed this for a plain `format!()` call, does the fix from #10055 also apply to this?

**Comment by taiki-e** [comment]
@djc I don't think it's a false positive. format_args_capture for `format!` is supported in all editions since Rust 1.58.0.
(This issue is about a bug that suggests using format_args_capture for `panic!`/`assert!`/`debug_assert!` that is not supported in the 2015/2018 edition.)

**Comment by djc** [comment]
That's fair, thanks!

**Comment by nyurik** [comment]
Should this be closed?

**Comment by nyurik** [comment]
@samueltardieu i think this is done?

**Comment by samueltardieu** [comment]
Looks like it, as `assert!(false, "foo is {}", hey);` will only trigger the lint on edition 2021 and later.

(side note: please don't ping me explicitly unless there is an obvious contextual reason to target me specifically [perceived availability is not one of them], if you need someone to react, then ping the whole Clippy team)
